src/core/utils.ts

```diff
@@ -453,3 +453,10 @@ export function getAbortController(): AbortController | undefined {
     return new AbortController()
   }
 }
+
+/**
+ * Type predicate to filter an array of null or undefined elements
+ */
+export function notNullOrUndefined<T>(value: T | null | undefined): value is T {
```

Comment:
we can simplify the types here:

```suggestion
export function notNullOrUndefined<T>(value: T): value is NonNullable<T> {
```